assets/js/plugin-check-admin.js

```diff
@@ -8,15 +8,21 @@
 		return;
 	}
 
-	checkItButton.addEventListener( 'click', (e) => {
-		e.preventDefault();
+	// Create a state object for the plugin checker.
+	const pluginChecker = {
+		plugin: false,
+		checks_to_run: [],
+	};
```

Comment:
Rather than storing data from the first AJAX response to send to use during the `runChecks` function, I wonder if we could pass these properties directly to the `runChecks` function as params?